### DISCUSSION TITLE
udp_connect vs udp_send(..., addr)

### DISCUSSION THREAD

**Discussion by dominictarr** [libuv/libuv#3436] [Q&A] [answered]
why is there both a `uv_udp_connect` method as well an `addr` argument to `uv_udp_send`?

the docs say:
> Associate the UDP handle to a remote address and port, so every message sent by this handle is automatically sent to that destination

that would make sense, if the handle was then used with `uv_write`, for example, but it doesn't mention that.
looking at the code, udp_send uses sendmsg - the docs for that https://linux.die.net/man/2/sendmsg
say
"The send() call may be used only when the socket is in a connected state (so that the intended recipient is known). The only difference between send() and write(2) is the presence of flags. With a zero flags argument, send() is equivalent to write(2)."

does that mean, if the uv_udp_connect has been called, then it's possible to just use `uv_write` to send messages? that is, to treat it just like a tcp stream?

**Comment by bnoordhuis** [comment, marked as answer]
`uv_udp_connect()` is purely a fast path. Connectedness doesn't really mean much with UDP sockets, it just lets you pass `NULL` as the destination address to `uv_udp_send()` and avoid some minor processing overhead.

You can't use `uv_write()` with UDP sockets.
